src/types/seat/seat.rs

```diff
@@ -256,9 +256,24 @@ wayland_listener!(Seat, (*mut wlr_seat, Box<SeatHandler>), [
 
         let drag_icon = DragIcon::new(data);
 
-        // TODO: call `new_drag_icon` on the handler and do something with the result
+        let (drag_icon_handler, surface_handler) =
+            handler.new_drag_icon(compositor, seat.weak_reference(), drag_icon.weak_reference());
+
+        if let Some(surface_handler) = surface_handler {
+            let surface_state = (*(*data).surface).data as *mut InternalSurfaceState;
+            (*(*surface_state).surface).data().1 = surface_handler;
+        }
+
+        if let Some(drag_icon_handler) = drag_icon_handler {
+            let mut listener = DragIconListener::new((drag_icon, drag_icon_handler));
+            wl_signal_add(&mut (*data).events.destroy as *mut _ as _,
+                          listener.destroy_listener() as _);
+            wl_signal_add(&mut (*data).events.map as *mut _ as _,
+                          listener.map_listener() as _);
```

Comment:
How do I keep it from dropping?

Reply:
`::std::mem::forget` [docs here](https://doc.rust-lang.org/beta/std/mem/fn.forget.html)

Reply:
Oh that's a pretty cool pattern. Why don't we do that everywhere?